src/parameterized_tendencies/microphysics/precipitation.jl

```diff
@@ -325,30 +325,32 @@ function compute_precipitation_surface_fluxes!(
     (; surface_rain_flux, surface_snow_flux) = p.precipitation
     (; col_integrated_precip_energy_tendency,) = p.conservation_check
     (; ᶜwᵣ, ᶜwₛ, ᶜspecific) = p.precomputed
+    ᶜlg = Fields.local_geometry_field(Y.c)
+    ˢlg = Fields.level(Fields.local_geometry_field(Y.f), Fields.half)
+    ⁱlg = Fields.Field(Fields.field_values(Fields.level(ᶜlg, 1)), axes(ˢlg))
 
-    (; ᶠtemp_scalar) = p.scratch
-    slg = Fields.level(Fields.local_geometry_field(ᶠtemp_scalar), Fields.half)
+    # Jacobian-weighted extrapolation from interior center to surface face,
+    # consistent with the 3D quantity ᶠρ = ᶠinterp(Y.c.ρ * ᶜJ) / ᶠJ 
+    ⁱρ = Fields.Field(Fields.field_values(Fields.level(Y.c.ρ, 1)), axes(ˢlg))
+    ˢρ = @. lazy(ⁱρ * ⁱlg.J / ˢlg.J)
```

Comment:
@trontrytel This change may be of interest to you.

Reply:
Per Zhaoyi's suggestion, the superscripts have been replaced with `sfc_` and `int_`.